src/libutil/affinity.cc

```diff
@@ -34,7 +42,13 @@ int lockToCurrentCPU()
 {
 #if __linux__
     int cpu = sched_getcpu();
-    if (cpu != -1) setAffinityTo(cpu);
+    // `lockToCurrentCPU` is re-entrant, because it is idempotent.
+    // But we have to avoid calling `setAffinity` twice,
+    // as that would corrupt the `savedAffinity`.
+    if (!didLockCPU && cpu != -1) {
+        didLockCPU = true;
+        setAffinityTo(cpu);
+    }
```

Comment:
I’m not sure I understand what the reason for this. Since `setAffinityTo` is pseudo-reentrant now, this code could stay unchanged (and we could avoid the need for the `didLockCPU` variable altogether), no?